Dockerfile

```diff
@@ -39,36 +46,37 @@ RUN mkdir -p /opt \
   && ln -s /opt/jdk1.${JAVA_VERSION_MAJOR}.0_${JAVA_VERSION_MINOR} ${JAVA_HOME}
 
 # install nexus
-ENV NEXUS_VERSION=3.0.2-02
-RUN mkdir -p /opt/sonatype/nexus \
+RUN mkdir -p ${NEXUS_HOME} \
   && curl --fail --silent --location --retry 3 \
     https://download.sonatype.com/nexus/3/nexus-${NEXUS_VERSION}-unix.tar.gz \
   | gunzip \
-  | tar x -C /opt/sonatype/nexus --strip-components=1 nexus-${NEXUS_VERSION} \
-  && chown -R root:root /opt/sonatype/nexus 
+  | tar x -C ${NEXUS_HOME} --strip-components=1 nexus-${NEXUS_VERSION} \
+  && chown -R root:root ${NEXUS_HOME}
 
-## configure nexus runtime env
-ENV NEXUS_CONTEXT='' \
-  NEXUS_DATA=/nexus-data
+# configure nexus
 RUN sed \
-    -e "s|karaf.home=.|karaf.home=/opt/sonatype/nexus|g" \
-    -e "s|karaf.base=.|karaf.base=/opt/sonatype/nexus|g" \
-    -e "s|karaf.etc=etc|karaf.etc=/opt/sonatype/nexus/etc|g" \
-    -e "s|java.util.logging.config.file=etc|java.util.logging.config.file=/opt/sonatype/nexus/etc|g" \
-    -e "s|karaf.data=data|karaf.data=${NEXUS_DATA}|g" \
-    -e "s|java.io.tmpdir=data/tmp|java.io.tmpdir=${NEXUS_DATA}/tmp|g" \
-    -i /opt/sonatype/nexus/bin/nexus.vmoptions \
+    -e "s|karaf.home=.|karaf.home=${NEXUS_HOME}|g" \
+    -e "s|karaf.base=.|karaf.base=${NEXUS_HOME}|g" \
+    -e "s|karaf.etc=etc|karaf.etc=${NEXUS_HOME}/etc|g" \
+    -e "s|java.util.logging.config.file=etc|java.util.logging.config.file=${NEXUS_HOME}/etc|g" \
+    -e "s|karaf.data=.*|karaf.data=${NEXUS_DATA}|g" \
+    -e "s|java.io.tmpdir=.*|java.io.tmpdir=${NEXUS_DATA}/tmp|g" \
+    -e "s|LogFile=.*|LogFile=${NEXUS_DATA}/log/jvm.log|g" \
+    -i ${NEXUS_HOME}/bin/nexus.vmoptions \
+  && mkdir -p ${NEXUS_DATA}/etc ${NEXUS_DATA}/log ${NEXUS_DATA}/tmp \
+  && cp ${NEXUS_HOME}/etc/nexus-default.properties ${NEXUS_DATA}/etc/nexus.properties \
   && sed \
     -e "s|nexus-context-path=/|nexus-context-path=/\${NEXUS_CONTEXT}|g" \
-    -i /opt/sonatype/nexus/etc/org.sonatype.nexus.cfg
+    -i ${NEXUS_DATA}/etc/nexus.properties
```

Comment:
Any reason not to make this change directly in nexus-default.properties, rather than making changes to what may be an external filesystem at runtime? If the user maps NEXUS_DATA then these changes will be lost.